infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -421,6 +414,14 @@ private LogReplicationSession constructSession(String sourceClusterId, String si
                 .build();
     }
 
+    private void generateNameForNewlyAddedSession() {
+        for(LogReplicationSession session : newSessionsDiscovered) {
+            String nameForSession = SESSION_NAME_PREFIX + sessionNameIdx;
```

Comment:
Would it be better to use a hash value suffix so the names are consistent on restarts or are the indexes persisted?

Reply:
I agree too Shama - on restarts this will end up with a different session mapping than from before restart right? 
Can we do something like this have a `hashCode()` implementation of the session which is generated from the string values of source, sink, model name , client name  fields?

Reply:
Sure, thats a good idea.
I'll start working on it